tests/integration_tests/platforms.py

```diff
@@ -0,0 +1,232 @@
+        if current_image:
+            return current_image
+        image_id = self.settings.OS_IMAGE
+        try:
+            image_id = self.client.released_image(self.settings.OS_IMAGE)
+        except (ValueError, IndexError):
+            pass
+        return image_id
+
+    def launch(self):
+        if self.settings.EXISTING_INSTANCE_ID:
+            print('Not launching instance due to EXISTING_INSTANCE_ID. '
+                  'Instance id: {}'.format(self.settings.EXISTING_INSTANCE_ID))
+            self.instance = self.client.get_instance(
+                self.settings.EXISTING_INSTANCE_ID
+            )
+            return
+        image_id = self._get_image()
+        launch_args = {
+            'image_id': image_id,
+            'user_data': self.user_data,
+            'wait': self.wait,
+        }
+        if self.instance_type:
+            launch_args['instance_type'] = self.instance_type
+        launch_args.update(self.launch_kwargs)
+        self.instance = self.client.launch(**launch_args)
+        print('Launched instance: {}'.format(self.instance))
+
+    def destroy(self):
+        self.instance.delete()
+
+    def exec(self, command):
+        return self.instance.execute(command)
+
+    def pull_file(self, remote_file, local_file):
+        self.instance.pull_file(remote_file, local_file)
+
+    def push_file(self, local_path, remote_path):
+        self.instance.push_file(local_path, remote_path)
+
+    def pull_from_file(self, remote_path) -> str:
+        tmp_file = NamedTemporaryFile('r')
+        self.pull_file(remote_path, tmp_file.name)
+        with tmp_file as f:
+            contents = f.read()
+        return contents
+
+    def push_to_file(self, remote_path, contents: str):
+        with NamedTemporaryFile('w', delete=False) as tmp_file:
+            tmp_file.write(contents)
+        self.push_file(tmp_file.name, remote_path)
+        os.unlink(tmp_file.name)
+
+    def snapshot(self):
+        return self.client.snapshot(self.instance, clean=True)
+
+    def _install_new_cloud_init(self, remote_script):
+        self.exec(remote_script)
+        version = self.exec('cloud-init -v').split()[-1]
+        print('Installed cloud-init version: {}'.format(version))
+        self.instance.clean()
+        image_id = self.snapshot()
+        print('Created new image: {}'.format(image_id))
+        set_current_image(image_id)
+        # self.instance.restart(wait=True)
+
+    def install_proposed_image(self):
+        print('Installing proposed image')
+        remote_script = (
+            '{sudo} echo deb "http://archive.ubuntu.com/ubuntu '
+            '$(lsb_release -sc)-proposed main" | '
+            '{sudo} tee /etc/apt/sources.list.d/proposed.list\n'
+            '{sudo} apt-get update -q\n'
+            '{sudo} apt-get install -qy cloud-init'
+        ).format(sudo='sudo' if self.use_sudo else '')
+        self._install_new_cloud_init(remote_script)
+
+    def install_ppa(self):
```

Comment:
Should this inspect an environment variable or cmdline param for which PPA to use. Currently it sets up proposed instead of installing a ppa.

Reply:
Isn't this how we install the dev-proposed PPA?

I was thinking mostly along the lines of when we SRU. Before it gets to proposed, we usually push a package to the dev-proposed PPA, right? How do we install that PPA?

Reply:
I was thinking more along the lines or making the PPA option a parameter that we can fill in explicitly. It'd give us more options if we want to test our own PPA for instance.
See my updated comment above.